docs/utils-reference/react-hooks/useCachedPromise.md

```diff
@@ -245,13 +245,32 @@ const { isLoading, data, pagination } = useCachedPromise(
 );
 ```
 
+or, if your data source uses cursor-based pagination, you can return a `cursor` alongside `data` and `hasMore`, and the cursor will be passed as an argument the next time the function gets called:
+
+```ts
+const { isLoading, data, pagination } = useCachedPromise(
+  (searchText: string) => async (options) => {
+    const response = await fetch(`https://api.example?q=${searchText}&cursor=${options.cursor}`);
+    const { data, nextCursor } = await response.json();
+    const hasMore = options.page < 50;
```

Comment:
maybe this should be `const hasMore = nextCursor !== undefined;`?

Reply:
good point!